src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,208 @@
+                "Registration should throw Custom exception"
+                        + " when login length is less than 6 characters");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void login_IsNull_NotOk() {
+        User user = new User();
+        user.setPassword("password");
+        user.setAge(33);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Login can't be null, login must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void password_IsNull_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setAge(33);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Password is empty, password must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void password_IsValid_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+        user.setAge(33);
+        registrationService.register(user);
+
+        assertTrue(user.getLogin().length() >= MIN_PASSWORD_LENGTH,
+                "Password must be at list 6 characters long");
+        User result = storageDao.get(user.getLogin());
+        assertEquals(user, result, "User must be added to storage");
+    }
+
+    @Test
+    void password_IsShort_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("pass");
+        user.setAge(33);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Password must be at list 6 characters long");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void password_IsLong_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("passwordislong");
+        user.setAge(33);
+
+        assertTrue(user.getLogin().length() >= MIN_PASSWORD_LENGTH,
+                "Password must be at list 6 characters long");
+        registrationService.register(user);
+        assertNotNull(storageDao.get(user.getLogin()), "User must be added to storage");
+    }
+
+    @Test
+    void age_IsNull_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "User age is null, must be minimum 18");
+        assertTrue(Storage.people.isEmpty(), "Storage must be empty");
+    }
+
+    @Test
+    void age_IsUnderRequirements_NotOk() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password");
+        user.setAge(17);
```

Comment:
To ensure consistent and efficient testing, it is recommended to set up a default valid user in the setup method and modify the relevant fields in corresponding test methods, such as changing the password to an invalid one